webui/react/src/components/kit/Facepile.tsx

```diff
@@ -0,0 +1,111 @@
+import { MinusOutlined, PlusOutlined } from '@ant-design/icons';
+import React, { useMemo, useState } from 'react';
+
+import SelectFilter from 'components/SelectFilter';
+import { DetailedUser } from 'types';
+
+import Button from './Button';
+import css from './Facepile.module.scss';
+import UserAvatar from './UserAvatar';
+
+export interface Props {
+  editable?: boolean;
+  selectableUsers?: DetailedUser[]; // This prop should be used to pass as options to the dropdown.
+  users?: DetailedUser[];
+}
+
+const Facepile: React.FC<Props> = ({ editable = false, selectableUsers = [], users = [] }) => {
+  const [showDropdown, setShowDropdown] = useState(false);
+  const [showAllAvatars, setShowAllAvatars] = useState(false);
+  const [avatars, setAvatars] = useState(users);
+  const amountOfAvatars = useMemo(() => avatars.length, [avatars.length]);
+  const usersItems = useMemo(
+    () =>
+      selectableUsers
+        .filter((user) => !avatars.find((av) => av.id === user.id))
+        .map((user) => ({
+          label: user.username,
+          value: user.id,
+        })),
+    [selectableUsers, avatars],
+  );
+  const visibleAvatars = useMemo(
+    () => (showAllAvatars ? avatars : avatars.slice(0, 5)),
+    [avatars, showAllAvatars],
+  );
+  const showButton = useMemo(() => editable || amountOfAvatars > 5, [editable, amountOfAvatars]);
+  const buttonLabel = useMemo(() => {
+    if (amountOfAvatars > 5 && !showAllAvatars) return `+ ${amountOfAvatars - 5}`;
+
+    return editable ? 'Add' : 'Hide';
```

Comment:
Buttons should be _just_ the icon, no text. Also you're using the plus icon for hide instead of the minus icon.